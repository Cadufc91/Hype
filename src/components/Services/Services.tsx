import * as React from 'react';
import {
    ServiceContainer,
    ServiceTitle,
    Divider1,
    CardsContainer,
    Cards,
    CardInfo,
    CardTitle,
    Divider2,
    CtaBtn,
}from './style';
import { BsArrowRight, BsBox, BsCode } from 'react-icons/bs';
import { FiLayers, FiColumns } from 'react-icons/fi';

const Services = () => {
    return(
        <ServiceContainer id='services'>
            <ServiceTitle>My Service</ServiceTitle>
            <Divider1 />
            <CardsContainer>
                <Cards>
                    <FiLayers size={40}/>
                    <CardInfo>
                        <CardTitle>UI/UX Design</CardTitle>
                        <Divider2 />
                        <CtaBtn>Discuss now</CtaBtn>
                        <BsArrowRight size={30} />
                    </CardInfo>
                </Cards>
                <Cards>
                    <BsBox  size={40}/>
                    <CardInfo>
                        <CardTitle>Product Design</CardTitle>
                        <Divider2 />
                        <CtaBtn>Discuss now</CtaBtn>
                        <BsArrowRight size={30} />

                    </CardInfo>
                </Cards>
                <Cards>
                    <FiColumns  size={40}/>
                    <CardInfo>
                        <CardTitle>Branding Design</CardTitle>
                        <Divider2 />
                        <CtaBtn>Discuss now</CtaBtn>
                        <BsArrowRight size={30} />

                    </CardInfo>
                </Cards>
                <Cards>
                    <BsCode  size={40}/>
                    <CardInfo>
                        <CardTitle>Front End Development</CardTitle>
                        <Divider2 />
                        <CtaBtn>Discuss now</CtaBtn>
                        <BsArrowRight size={30} />

                    </CardInfo>
                </Cards>
            </CardsContainer>
        </ServiceContainer>
    )
};

export default Services;