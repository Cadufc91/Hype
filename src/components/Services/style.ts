import styled from "styled-components";

export const ServiceContainer = styled.div`
    margin-top: 67px;
    margin-bottom: 80px;
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 468px;
`;

export const ServiceTitle = styled.h1`
    font-size: 40px;
    font-weight: 400;
    color: #fff;
`;

export const Divider1 = styled.span`
    width: 80px;
    height: 3px;
    background: #ED1CA6;
    border: 1px solid #ED1CA6;
    margin-bottom: 50px;
`;

export const CardsContainer = styled.div`
    display: flex;
    flex-direction: row;
    max-width: 70%;
    height: 351px;
`;

export const Cards = styled.div`
    padding: 22px;
    width: 400px !important;
    background: #fff;
    margin-right: 24px;
    border-radius: 6px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    :hover{
        color: #ED1CA6;
    }
`;

export const CardInfo = styled.div`
    display: flex;
    flex-direction: column;
    
`;

export const CardTitle = styled.h2`
    margin-top: 100px;
    margin-bottom: 16px;
    font-size: 32px;
    font-weight: 400;
`;

export const Divider2 = styled.span`
    width: 64px;
    height: 3px;
    background: #ED1CA6;
    border: 1px solid #ED1CA6;
    border-radius: 50px;
    opacity: 30%;
    z-index: 1;
    margin-bottom: 24px;
`;

export const CtaBtn = styled.p`
    display: none;
`;
