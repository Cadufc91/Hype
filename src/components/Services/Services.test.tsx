import { render } from "@testing-library/react";
import  { Services } from './Services';

describe('Header', () => {
    it('should match snapshot', () => {
        const { container } = render(<Services />)

        expect(container.firstChild).toMatchSnapshot()
    })
})