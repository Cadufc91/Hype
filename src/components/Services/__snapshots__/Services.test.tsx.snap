// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div
  class="sc-bczRLJ bsMIqx"
  id="services"
>
  <h1
    class="sc-gsnTZi gSiNDo"
  >
    Servi√ßos
  </h1>
  <span
    class="sc-dkzDqf jdJdDv"
  />
  <div
    class="sc-hKMtZM fvJiXf"
  >
    <div
      class="sc-eCYdqJ jmxKtw"
    >
      <svg
        fill="none"
        height="40"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="12 2 2 7 12 12 22 7 12 2"
        />
        <polyline
          points="2 17 12 22 22 17"
        />
        <polyline
          points="2 12 12 17 22 12"
        />
      </svg>
      <div
        class="sc-jSMfEi bHwFYj"
      >
        <h2
          class="sc-gKXOVf kMABnC"
        >
          UI/UX Design
        </h2>
        <span
          class="sc-iBkjds cRVuwR"
        />
        <p
          class="sc-ftvSup jDzSUa"
        >
          Conversar
        </p>
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCYdqJ jmxKtw"
    >
      <svg
        fill="currentColor"
        height="40"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.186 1.113a.5.5 0 0 0-.372 0L1.846 3.5 8 5.961 14.154 3.5 8.186 1.113zM15 4.239l-6.5 2.6v7.922l6.5-2.6V4.24zM7.5 14.762V6.838L1 4.239v7.923l6.5 2.6zM7.443.184a1.5 1.5 0 0 1 1.114 0l7.129 2.852A.5.5 0 0 1 16 3.5v8.662a1 1 0 0 1-.629.928l-7.185 2.874a.5.5 0 0 1-.372 0L.63 13.09a1 1 0 0 1-.63-.928V3.5a.5.5 0 0 1 .314-.464L7.443.184z"
        />
      </svg>
      <div
        class="sc-jSMfEi bHwFYj"
      >
        <h2
          class="sc-gKXOVf kMABnC"
        >
          Product Design
        </h2>
        <span
          class="sc-iBkjds cRVuwR"
        />
        <p
          class="sc-ftvSup jDzSUa"
        >
          Conversar
        </p>
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCYdqJ jmxKtw"
    >
      <svg
        fill="none"
        height="40"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3h7a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-7m0-18H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h7m0-18v18"
        />
      </svg>
      <div
        class="sc-jSMfEi bHwFYj"
      >
        <h2
          class="sc-gKXOVf kMABnC"
        >
          Branding Design
        </h2>
        <span
          class="sc-iBkjds cRVuwR"
        />
        <p
          class="sc-ftvSup jDzSUa"
        >
          Conversar
        </p>
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCYdqJ jmxKtw"
    >
      <svg
        fill="currentColor"
        height="40"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.854 4.854a.5.5 0 1 0-.708-.708l-3.5 3.5a.5.5 0 0 0 0 .708l3.5 3.5a.5.5 0 0 0 .708-.708L2.707 8l3.147-3.146zm4.292 0a.5.5 0 0 1 .708-.708l3.5 3.5a.5.5 0 0 1 0 .708l-3.5 3.5a.5.5 0 0 1-.708-.708L13.293 8l-3.147-3.146z"
        />
      </svg>
      <div
        class="sc-jSMfEi bHwFYj"
      >
        <h2
          class="sc-gKXOVf kMABnC"
        >
          Front End Development
        </h2>
        <span
          class="sc-iBkjds cRVuwR"
        />
        <p
          class="sc-ftvSup jDzSUa"
        >
          Conversar
        </p>
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
